internal/hardware/validator.go

```diff
@@ -36,14 +36,15 @@ func NewValidator(log logrus.FieldLogger, cfg ValidatorCfg, operatorsAPI operato
 }
 
 type ValidatorCfg struct {
-	MinCPUCores                   int64 `envconfig:"HW_VALIDATOR_MIN_CPU_CORES" default:"2"`
-	MinCPUCoresWorker             int64 `envconfig:"HW_VALIDATOR_MIN_CPU_CORES_WORKER" default:"2"`
-	MinCPUCoresMaster             int64 `envconfig:"HW_VALIDATOR_MIN_CPU_CORES_MASTER" default:"4"`
-	MinRamGib                     int64 `envconfig:"HW_VALIDATOR_MIN_RAM_GIB" default:"8"`
-	MinRamGibWorker               int64 `envconfig:"HW_VALIDATOR_MIN_RAM_GIB_WORKER" default:"8"`
-	MinRamGibMaster               int64 `envconfig:"HW_VALIDATOR_MIN_RAM_GIB_MASTER" default:"16"`
-	MinDiskSizeGb                 int64 `envconfig:"HW_VALIDATOR_MIN_DISK_SIZE_GIB" default:"120"` // Env variable is GIB to not break infra
-	MaximumAllowedTimeDiffMinutes int64 `envconfig:"HW_VALIDATOR_MAX_TIME_DIFF_MINUTES" default:"4"`
+	MinCPUCores                   int64                        `envconfig:"HW_VALIDATOR_MIN_CPU_CORES" default:"2"`
```

Comment:
The new `VersionedRequirements` doesn't replace the other variables?

Reply:
No; for the backwards compatibility and providing last-resort defaults (when version is not in JSON).

Reply:
I'm not sure we need to support backward compatibility for deployments. Aligning to one approach would be simpler. Maybe for default, we could have a "default" entry in the JSON?

Reply:
That makes perfect sense and I would tackle it in three phases to make the transition smooth:
- introduction of the new mechanism without removing the old one (this PR);
- switching all the external usages (tests?) of the old mechanism to the new one (where would you recommend me to look?);
- removing the old mechanism (another PR);

WDYT?

Reply:
Sure that makes sense.
AFAIK only `assisted-service` configmaps and yamls, `assisted-test-infra`, and the `saas.yaml` in the cloud. Please post about this transition on the Slack channel to verify if there are other places involved.